SP 611: Create Word Links directory and allow CSV files to have different names

DaleHensley opened the pull request:
Per Issue #611, update Story Producer to create the Word Links Directory if it does not exist and allow the CSV file that contains the word links data to have a variable name.  This name begins with "wordlinks" and ends with ".csv", e.g.: worklinks-DKH.csv.

Testing:  This fix was successfully tested on an Android 12, Pixel 5 phone and on an Android 8, Pixel 3 Android Studio emulation.

bryanwussow commented on app/src/main/java/org/sil/storyproducer/model/Workspace.kt:
```diff
@@ -26,10 +27,13 @@ internal const val DEMO_FOLDER = "000 Unlocked demo story Storm"
 internal const val WORD_LINKS_JSON_FILE = "wordlinks.json"
 internal const val WORD_LINKS_CLICKED_TERM = "ClickedTerm"
 internal const val WORD_LINKS_SLIDE_NUM = "CurrentSlideNum"
+// DKH - 11/19/2021 Issue #661 Allow CSV file to have different names
```
#611

bryanwussow commented on app/src/main/java/org/sil/storyproducer/model/Workspace.kt:
```diff
@@ -152,12 +156,34 @@ object Workspace {
             mapTermFormsToTerms()
             buildWLSTree()
+        }else{
+            // DKH - 11/19/2021 Issue #661 Create Word Links CSV directory if it does not exist
```
#611

bryanwussow commented on app/src/main/java/org/sil/storyproducer/model/Workspace.kt:
```diff
@@ -152,12 +156,34 @@ object Workspace {
     private fun importWordLinksFromCSV(context: Context, wordLinksDir: DocumentFile){
-        val wordLinksFile = wordLinksDir.findFile(WORD_LINKS_CSV_FILE)
-        if (wordLinksFile != null) {
+        // DKH - 11/19/2021 Issue #661 Allow CSV file to have different names
```
#611

bryanwussow reviewed (commented):
Code looks good, but some issue# typos in your comments might be confusing in the future  :-/

DaleHensley commented:
Thanks for the feedback.  I updated those comments for SP 613 (autoload
wordlinks file), so, if you would go ahead and process SP 611 (pull 612),
those comments will be fixed when I turn over SP613.  This will make it
easier for me to do the merges into GIT because I built SP613 off the
current version of 611.

For SP615, I added comments in the issues page.
Dale


On Wed, Dec 8, 2021 at 4:52 PM Bryan Wussow ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
>
> Code looks good, but some issue# typos in your comments might be confusing
> in the future :-/
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/sillsdev/StoryProducer/pull/612#pullrequestreview-827100526>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ASW7JVUJUF6TT267YXIV4MDUP74V3ANCNFSM5IONKKYQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>

bryanwussow reviewed (approved):
👍
